src/mlx/warnings/regex_checker.py

```diff
@@ -107,12 +161,156 @@ def check(self, content):
         '''
         matches = re.finditer(self.pattern, content)
         for match in matches:
-            if (match.group('curr') == match.group('max')) and \
-                    (match.group('classification') in self.CLASSIFICATION):
-                self.count += 1
-                match_string = match.group(0).strip()
-                self.counted_warnings.append(match_string)
-                self.print_when_verbose(match_string)
+            if (classification := match.group("classification").lower()) in self.checkers:
+                self.checkers[classification].check(match)
+            else:
+                checker = CoverityClassificationChecker(classification=classification, verbose=self.verbose)
+                self.checkers[classification] = checker
+                checker.cq_enabled = self.cq_enabled
+                checker.exclude_patterns = self.exclude_patterns
+                checker.cq_description_template = self.cq_description_template
+                checker.cq_default_path = self.cq_default_path
+                checker.check(match)
+
+    def parse_config(self, config):
+        """Process configuration
+
+        Args:
+            config (dict): Content of configuration file
+        """
+        for key in config:
```

Comment:
Before the `for key in config:`, you could process all non-classification keys whilst removing them with `dict.pop`.

```
        if value := config.pop("cq_description_template", None)
            self.cq_description_template = Template(value)
        for key in config:
            classification_key = key.replace("_", " ").lower()
            if classification_key in CoverityClassificationChecker.VERBOSITY_MAP:
                checker = CoverityClassificationChecker(classification_key, verbose=self.verbose)
                ...
            else:
                print(f"WARNING: Unrecognized classification {key!r}")
```

Reply:
Resolved in c5b72edf6a3d14cbc673ce0ef215244874d50dbe

Reply:
Are you not in favor of using `CoverityClassificationChecker.VERBOSITY_MAP` here?

Reply:
I find `VERBOSITY_MAP` confusing if these contain `["unclassified", "pending", "false positive", "intentional", "bug"]`.
Or do you mean something else?

Reply:
Sorry, I meant `CoverityClassificationChecker.SEVERITY_MAP`

Reply:
No problem, resolved in f1d84bd8052b61cf91dea2ce4080d1105af24ee6